src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCookieHelper.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    internal static class WinHttpCookieHelper
```

Comment:
This doesn't really feel like a static class: most functions keep getting the state object. Why not construct a new object instead.

Also, the name isn't conforming to the single responsibility principle: in my mind a "helper" is good for anything so it'll end up getting all sorts of functions doing various things with cookies over the time. The actual purpose seems to be adapting from WinHttp to .Net CookieContainer: maybe renaming to CookieContainerAdapter would make the intent clear?